src/org/webonise/multithreading/concurrentmodificationresolution/WorkerThread.java

```diff
@@ -0,0 +1,24 @@
+package org.webonise.multithreading.concurrentmodificationresolution;
+
+import java.util.List;
+
+public class WorkerThread implements Runnable {
+    private List<Integer> list;
+
+    WorkerThread(List<Integer> list){
+        this.list = list;
+    }
+
+    @Override
+    public void run() {
+        while(!list.isEmpty()){
+            list.remove(list.size() - 1);
```

Comment:
Yes we can use iterator for removing but not in this case. This code is for resolution of concurrent modification exception using copyonwritearraylist. iterator of this arraylist do not support remove, set and add operations. Doing so will throw UnsupportedOperationException. hence list.remove is used. please refer this java doc more detailed information : https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/CopyOnWriteArrayList.html